### ISSUE TITLE
missing State Registration document

### ISSUE THREAD

**Issue by tonicospinelli** [brazanation/documents#16]
State Registration is so complex, because it has different numbers based on State rules.

So, what are states from Brazil?
- [x] AC - ACRE
- [x] AL - ALAGOAS
- [x] AM - AMAZONAS
- [x] AP - AMAPÁ
- [x] BA - BAHIA
- [x] CE - CEARÁ
- [x] DF - DISTRITO FEDERAL
- [x] ES - ESPÍRITO SANTO
- [x] GO - GOIÁS
- [x] MA - MARANHÃO
- [x] MG - MINAS GERAIS
- [x] MS - MATO GROSSO DO SUL
- [x] MT - MATO GROSSO
- [x] PA - PARÁ
- [x] PB - PARAÍBA
- [x] PE - PERNAMBUCO
- [x] PI - PIAUÍ
- [x] PR - PARANÁ
- [x] RJ - RIO DE JANEIRO
- [x] RN - RIO GRANDE DO NORTE
- [x] RO - RONDÔNIA
- [x] RR - RORAIMA
- [x] RS - RIO GRANDE DO SUL
- [x] SC - SANTA CATARINA
- [x] SE - SERGIPE
- [x] SP - SÃO PAULO
- [x] TO - TOCANTINS

**Comment by Ferreiramg** [comment]
I suggest implementing on IE namespace , class representing each state. And later one factory create as instantiator .

`$iesp = new IE\SP($number);`

**Comment by tonicospinelli** [comment]
@Ferreiramg I have a similar approach, but does to not using namespace, is using magic call `__callStatic` method to create object based on State.

``` php
$ie = new StateRegistration($number, StateRegistration::SP);

// or with facade

$ie = StateRegistration::SP($number);
```

**Comment by tonicospinelli** [comment]
fixed by #19
